packages/utils/telemetry-utils/src/logger.ts

```diff
@@ -531,6 +542,16 @@ export class PerformanceEvent {
 		event.eventName = `${event.eventName}_${eventNameSuffix}`;
 		if (eventNameSuffix !== "start") {
 			event.duration = this.duration;
+			if (this.startMemory) {
+				const currentMemory = (performance as PerformanceWithMemory)?.memory
+					?.usedJSHeapSize;
+				const differenceInKBytes = Math.floor((currentMemory - this.startMemory) / 1024);
+				if (differenceInKBytes > 0) {
+					event.usedJSHeapSize = differenceInKBytes;
+				}
+			}
+		} else if (this.collectMemory) {
+			this.startMemory = (performance as PerformanceWithMemory)?.memory?.usedJSHeapSize;
```

Comment:
what is the behavior if cross-origin isolation is not enabled? The docs say this throws? https://developer.mozilla.org/en-US/docs/Web/API/Performance/measureUserAgentSpecificMemory#exceptions

Should we add a check for that before calling the API?

Reply:
I didn't read the description with the required attention, sorry.